### PR TITLE
content

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: mule-sdk/v/4.0/content_parameters.adoc [hunk 1]
```diff
@@ -1 +1,182 @@
...
+There’re edge cases in which a primary content should default to something else than the payload. This tends to happen more when the operation has only one content 
+parameter and that parameter is not always needed. For example consider the Database connector. The input parameters of a select query are primary content, but not 
+all queries require input parameters. So the real default here should be an emptyMap
+
+You can do that by combining the `@Content` and `@Optional` annotations. For example, the following sets the default to an empty map:
+
+[source, java, linenums]
+----
+public List<Map> select(@Text String sql,
+                        @Optional(defaultValue="#[[]]") @Content Map<String, Object> inputParameters) {
```
Comment (estebanwasinger) [inline mule-sdk/v/4.0/content_parameters.adoc]: wrong expression `[]` -> `{}`

File: mule-sdk/v/4.0/content_parameters.adoc [hunk 2]
```diff
@@ -1 +1,182 @@
...
+----
+public List<Map> select(@Text String sql,
+                        @Optional @NullSafe @Content Map<String, Object> inputParameters) {
+	// select
+}
+----
+
+[TIP]
+Examples 1 and 3 are perfectly equivalent. The `@NullSafe` option is preferred though since it gives the 
+module's user a better experience. Having `#[[]]` as an explicit default may be confusing for non experienced
```
Comment (estebanwasinger) [inline mule-sdk/v/4.0/content_parameters.adoc]: same here

File: mule-sdk/v/4.0/content_parameters.adoc [hunk 3]
```diff
@@ -1 +1,182 @@
...
+[source, xml, linenums]
+----
+<file:copy from="somePath" to="anotherPath" />
+----
+
+Both of these parameters are behavior ones since this operation doesn’t take any content. The content is in the file being copied.
+
+== Content Parameters
+
+Considering the explained above, the definition of content parameter is at this point obvious. In the `file:write` example, the content is a content parameter.
```
Comment (estebanwasinger) [inline mule-sdk/v/4.0/content_parameters.adoc]: `the definition of content parameter` -> `the definition of a content parameter`

File: mule-sdk/v/4.0/content_parameters.adoc [hunk 4]
```diff
@@ -1 +1,182 @@
...
+
+* They’re automatically made optional
+* They automatically default to `#[payload]`
+
+These two features are automatically added to the parameter by the runtime. 
+
+[TIP]
+Going back to the `file:write` example in which there was only one content parameter: is it primary? _YES_. When an operation has only one content parameter, 
+the SDK will automatically consider it as primary. That means that the content parameter in the `file:write` operation is automatically made optional and defaults
+to `#[payload]`. This helps enforcing consistency across modules.
```
Comment (estebanwasinger) [inline mule-sdk/v/4.0/content_parameters.adoc]: `This helps enforcing` -> `This helps to enforce`

File: mule-sdk/v/4.0/content_parameters.adoc [hunk 5]
```diff
@@ -1 +1,182 @@
...
+
+== Changing the default of a primary content
+
+There’re edge cases in which a primary content should default to something else than the payload. This tends to happen more when the operation has only one content 
+parameter and that parameter is not always needed. For example consider the Database connector. The input parameters of a select query are primary content, but not
```
Comment (estebanwasinger) [inline mule-sdk/v/4.0/content_parameters.adoc]: `For example consider` -> `For example, consider`

File: mule-sdk/v/4.0/content_parameters.adoc [hunk 6]
```diff
@@ -1 +1,182 @@
...
+[TIP]
+Going back to the `file:write` example in which there was only one content parameter: is it primary? _YES_. When an operation has only one content parameter, 
+the SDK will automatically consider it as primary. That means that the content parameter in the `file:write` operation is automatically made optional and defaults
+to `#[payload]`. This helps enforcing consistency across modules.
+
+== Changing the default of a primary content
+
+There’re edge cases in which a primary content should default to something else than the payload. This tends to happen more when the operation has only one content 
+parameter and that parameter is not always needed. For example consider the Database connector. The input parameters of a select query are primary content, but not 
+all queries require input parameters. So the real default here should be an emptyMap
```
Comment (estebanwasinger) [inline mule-sdk/v/4.0/content_parameters.adoc]: `emptyMap` -> `empty map` ?

File: mule-sdk/v/4.0/content_parameters.adoc [hunk 7]
```diff
@@ -1 +1,182 @@
...
+
+[TIP]
+Examples 1 and 3 are perfectly equivalent. The `@NullSafe` option is preferred though since it gives the 
+module's user a better experience. Having `#[{}]` as an explicit default may be confusing for non experienced 
+Mule users.
+
+== Embedding content parameters in parameter groups
+
+Looking closely at the `http:request` operation, you’ll see that the content parameters are contained 
+in an element called `request-builder`. That’s because for usability reasons, the author of the
```
Comment (estebanwasinger) [inline mule-sdk/v/4.0/content_parameters.adoc]: `That’s because for` -> `That’s because, for`

File: mule-sdk/v/4.0/content_parameters.adoc (hunk 3 above)
Comment (alegmarra) [reply mule-sdk/v/4.0/content_parameters.adoc]: "the `content` is a `@Content` parameter"

**Pull request merged**